packages/cli/src/commands/export-dynamic-plugin/backend.ts

```diff
@@ -0,0 +1,320 @@
+    customizeManifest: (pkgToCustomize: BackstagePackageJson) => {
+      function test(str: string, expr: string | RegExp): boolean {
+        if (typeof expr === 'string') {
+          return str === expr;
+        }
+        return expr.test(str);
+      }
+
+      pkgToCustomize.name = `${pkgToCustomize.name}-dynamic`;
+      (pkgToCustomize as any).bundleDependencies = true;
+      pkgToCustomize.scripts = undefined;
+      pkgToCustomize.types = undefined;
+
+      pkgToCustomize.files = pkgToCustomize.files?.filter(
+        f => !f.startsWith('dist-dynamic/'),
+      );
+
+      for (const dep in dependenciesToAdd) {
+        if (!Object.prototype.hasOwnProperty.call(dependenciesToAdd, dep)) {
+          continue;
+        }
+        if (pkgToCustomize.dependencies === undefined) {
+          pkgToCustomize.dependencies = {};
+        }
+        const existingVersion = pkgToCustomize.dependencies[dep];
+        if (existingVersion === undefined) {
+          pkgToCustomize.dependencies[dep] = dependenciesToAdd[dep];
+          continue;
+        }
+        if (existingVersion !== dependenciesToAdd[dep]) {
+          if (!semver.satisfies(existingVersion, dependenciesToAdd[dep])) {
+            throw new Error(
+              `The version of a dependency ('${dep}') of an embedded module conflicts with main module dependencies: '${dependenciesToAdd[dep]}', '${existingVersion}'`,
+            );
+          }
+        }
+      }
+      if (pkgToCustomize.dependencies) {
+        for (const monoRepoPackage of monoRepoPackages.packages) {
+          if (pkgToCustomize.dependencies[monoRepoPackage.packageJson.name]) {
+            pkgToCustomize.dependencies[
+              monoRepoPackage.packageJson.name
+            ] = `^${monoRepoPackage.packageJson.version}`;
+          }
+        }
+
+        for (const dep in pkgToCustomize.dependencies) {
+          if (
+            !Object.prototype.hasOwnProperty.call(
+              pkgToCustomize.dependencies,
+              dep,
+            )
+          ) {
+            continue;
+          }
+          let removed = false;
+          for (const toRemove of mergeWithOutput) {
+            if (test(dep, toRemove)) {
+              delete pkgToCustomize.dependencies[dep];
+              removed = true;
+              break;
+            }
+          }
+          if (removed) {
+            continue;
+          }
+
+          for (const toMove of moveToPeerDependencies) {
+            if (test(dep, toMove)) {
+              console.log(`Moving '${dep}' to peerDependencies`);
+
+              if (pkgToCustomize.peerDependencies === undefined) {
+                pkgToCustomize.peerDependencies = {};
+              }
+              pkgToCustomize.peerDependencies[dep] =
+                pkgToCustomize.dependencies[dep];
+              delete pkgToCustomize.dependencies[dep];
+              break;
+            }
+          }
+        }
+      }
+      pkgToCustomize.devDependencies = {};
```

Comment:
Why do we reset this? `devDependencies` are not bundled anyways. If you're trying to avoid `devDependencies` during `yarn install`, simply run `yarn install --production`

https://classic.yarnpkg.com/lang/en/docs/cli/install/#toc-yarn-install-production-true-false

Reply:
This package has no sources, and no scripts in it. It is already derived from the source package. I don't see how it makes sense to add `devDependencies`.

Reply:
I added back `devDependencies` in the derived package, as discussed.

Reply:
In fact I removed it back again since, at least in yarn v1 `yarn --production` still brings the `devDependencies`, at least the first time it is run, and there's no way to remove the dev dependencies from the generated `yarn.lock`.

See https://github.com/yarnpkg/yarn/issues/6373#issuecomment-760068356